### PR TITLE
Customad1

### PR DESCRIPTION
**Pull request by xgretsch**
When an ad was not present, an attempt was still being made to find and
render the template, which was causing errors.

Documentation now has an
example on how to use amp-analytics, and removed the wrong line about
problems with image layout.

### CODE REVIEW

File: extensions/amp-ad/0.1/amp-ad-custom.js [hunk 1]
```diff
@@ -96,17 +96,19 @@ export class AmpAdCustom extends AMP.BaseElement {
...
-        });
+      if (templateData !== null && typeof templateData == 'object') {
+        this.uiHandler.setDisplayState(AdDisplayState.LOADED_RENDER_START);
+        templatesFor(this.win).findAndRenderTemplate(element, templateData)
+          .then(renderedElement => {
+          // Get here when the template has been rendered
+          // Clear out the template and replace it by the rendered version
+            removeChildren(element);
+            element.appendChild(renderedElement);
+            return this;
```
Comment (jridgewell) [inline extensions/amp-ad/0.1/amp-ad-custom.js]: Is there a need for this return?

Comment (xgretsch) [inline extensions/amp-ad/0.1/amp-ad-custom.js]: I'm sure it was requested during a code review at some point, but I can't find it in the comment trail, I've done much head-scratching and I can't remember why. The code appears to work fine without it. Presumably we should take it out - unless @lannka or someone else can remember why it's there! Let me know, or feel free to delete line 107 before merging.

File: ads/custom.md [hunk 2]
```diff
@@ -172,15 +166,52 @@ If no slot was specified, the server returns a single template rather than an ar
...
+```html
+<amp-ad type="custom" layout="responsive" width="300" height="250" 
+    data-url="https://mysite/my-ad-server">
+    <template type="amp-mustache" id="my-amp-template-id">
+        <a href="{{href}}" data-vars-event-label="{{evehtLabel}}" data-vars-event-category="{{category}}">
+            <amp-img layout='responsive' width="300" height="250" src="{{artwork}}"></amp-img>
+        </a>
+    </template>  
+</amp-ad>
+<amp-analytics type='googleanalytics' id='myamppage'>
```
Comment (lannka) [inline ads/custom.md]: i guess `id` is not needed?

File: extensions/amp-ad/0.1/amp-ad-custom.js (hunk 1 above)
Comment (lannka) [inline extensions/amp-ad/0.1/amp-ad-custom.js]: let's remove it

Comment (xgretsch) [inline extensions/amp-ad/0.1/amp-ad-custom.js]: I guess the "allow maintainers to make changes" isn't as useful as I hoped! I've just done it.

File: ads/custom.md (hunk 2 above)
Comment (xgretsch) [inline ads/custom.md]: Probably not, I've got rid of it.

**Pull request merged**